startup/81-dexela.py

```diff
@@ -0,0 +1,72 @@
+import os
+import ophyd
+from hxntools.detectors.dexela import (DexelaDetector, DexelaTiffPlugin)
+from ophyd import (AreaDetector, CamBase, TIFFPlugin, Component as Cpt,
+                    HDF5Plugin, Device, StatsPlugin, ProcessPlugin,
+                    ROIPlugin, EpicsSignal)
+from databroker.assets.handlers import HandlerBase
+from ophyd.areadetector.filestore_mixins import (FileStoreIterativeWrite,
+                                                 FileStoreHDF5IterativeWrite,
+                                                 FileStoreTIFFSquashing,
+                                                 FileStoreTIFF,
+                                                 FileStoreHDF5, new_short_uid,
+                                                 FileStoreBase
+                                                 )
+from ophyd.areadetector import (AreaDetector, PixiradDetectorCam, ImagePlugin,
+                                TIFFPlugin, StatsPlugin, HDF5Plugin,
+                                ProcessPlugin, ROIPlugin, TransformPlugin,
+                                OverlayPlugin)
+from ophyd.areadetector.trigger_mixins import SingleTrigger
+from enum import Enum
+
+
+class XPDDMode(Enum):
+    step = 1
+    fly = 2
+
+class XPDDDexelaTiffPlugin(TIFFPlugin, FileStoreTIFF, Device):
+    pass
+
+class XPDDDexelaDetector(SingleTrigger, DexelaDetector):
+    total_points = Cpt(Signal, value=1, doc="The total number of points to be taken")
+    tiff = Cpt(XPDDDexelaTiffPlugin, 'TIFF1:',
+               read_attrs=[],
+               configuration_attrs=[],
+               write_path_template='Z:dex_data\\',
```

Comment:
I think the right path would be:
```suggestion
               write_path_template='Z:\\dex_data\\%Y\\%m\\%d\\',
```
so that it matches the read path.

Reply:
I'm not quiet sure. maybe eventually we need configure it to use your path, at least has \\%Y\\%m\\%d\\. However, current path let me go through path error.

Reply:
The point here is that the detector will write to a different location than you read the data from, so the write and read paths should be coordinated. Otherwise the collected files will appear in `Z:dex_data` folder (an equivalent path on Linux `/nsls2/xf28id2/dex_data/`), so databroker will contain records with broken file references.

Reply:
@ke-zhang-rd, please consider committing the suggestion via the <kbd>Commit suggestion</kbd> button.